src/main/java/com/google/devtools/build/lib/actions/SpawnResult.java

```diff
@@ -260,16 +260,30 @@ default String getFailureMessage() {
    *
    * <p>This behavior may be triggered with {@link
    * ExecutionRequirements#REMOTE_EXECUTION_INLINE_OUTPUTS}.
+   *
+   * <p>Implementing classes should override {@link #getInMemoryOutputBytes(ActionInput)} instead.
    */
   @Nullable
   default InputStream getInMemoryOutput(ActionInput output) {
+    ByteString bytes = getInMemoryOutputBytes(output);
+    if (bytes == null) {
+      return null;
+    }
+    return bytes.newInput();
+  }
```

Comment:
On the other hand: do we even need to change the contract? Couldn't we just as easily delegate to an `InputStream` instead of a` ByteString` in `RemoteExecutionService.waitForAndReuseOutputs`? Or am I missing something?

Reply:
Oh, it's because an `InputStream` can only be read once, and we need to be able to reuse multiple times.

Reply:
That led to a nice cleanup, thanks!